### PR TITLE
feat: updates for winning and window post, and sector shapes 

### PR DESCRIPTION
**Pull request by dignifiedquire**
Depends on https://github.com/filecoin-project/rust-fil-proofs/pull/1075

### CODE REVIEW

File: src/post.rs [hunk 1]
```diff
@@ -1,107 +1,219 @@
...
-    let candidates_v1 = filecoin_proofs_v1::generate_candidates(
-        config_v1.expect("checked before").as_v1_config(),
+) -> Result<Vec<SectorId>> {
+    ensure!(
+        proof_type.typ() == PoStType::Winning,
+        "invalid post type provide"
+    );
+
+    with_shape!(
+        u64::from(proof_type.sector_size()),
```
Comment (porcuquine) [inline src/post.rs:24]: @dignifiedquire Why does this work? This looks like a runtime call to get the sector size. How will that be resolved at macroexpansion time?

Comment (dignifiedquire) [reply src/post.rs:24]: `with_shape!` is runtime expansion, it uses a match statement to go from a runtime sector size to a compile time type of the actual sector shape, iff there is a match.

Comment (porcuquine) [reply src/post.rs:24]: Got it.

**Comment by porcuquine** [comment]
Based on conversation, I'm fine with merging without 64GiB if this will be addressed in follow-up.

**Pull request merged**